### PR TITLE
Finish exercise

### PR DESCRIPTION
**Pull request by lsaville**
@jcasimir Yo

### CODE REVIEW

**Comment by jcasimir** [comment]
👍

File: lib/string_validator.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require 'pry'
+
+class StringValidator
+  def validate(string)
+    stack = []
+    string.split('').each do |character|
```
Comment (mikedao) [inline lib/string_validator.rb:6]: It looks like you're building an array here. Can we avoid the use of the collector local variable by using a different enumerable, such as `reduce/inject`?

File: lib/string_validator.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      if library.values.include?(character)
+        stack << character
+      elsif library[character]
+        stack.pop
+      end
+    end
+
+    if stack == []
```
Comment (mikedao) [inline lib/string_validator.rb:14]: This bit feels a bit tacked on, perhaps we should pull it out to its own method?

File: spec/string_validator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+require 'string_validator'
+
+RSpec.describe StringValidator do
```
Comment (mikedao) [inline spec/string_validator_spec.rb:3]: WTF Rspec? 

Kidding.

Comment (mikedao) [reply spec/string_validator_spec.rb:3]: Do remember to add in your sad path tests and edge case tests. They may not be one hundred percent useful, but it shows people that you're thinking ahead and thinking about possibilities. It makes people assume that you're a lot deeper than you may be.

**Review by mikedao — APPROVED** [review]
